bancor_research/bancor_simulator/v3/spec/network.py

```diff
@@ -187,8 +191,9 @@ def deposit(
         Top level logic for deposit actions.
         """
         state = self.get_state(copy_type="initial", timestamp=timestamp)
-        state, tkn_name, user_name = validate_input(
-            state, tkn_name, user_name, timestamp
+        tkn_amt = userAmount(state, tkn_name, user_name, tkn_amt)
```

Comment:
How/why would `tkn_amt` be computed based on `state, tkn_name, user_name`? This is a `BancorDapp` method, meaning it is the intended API for users of this codebase - hence they should be the ones specifying what the `tkn_amt` is. The deleted `validate_input(..)` did not include `tkn_amt` per your recent changes, presumably because the validation would have failed for string amounts with `%` signs, however, it makes more sense to simply call `validate_input(..)` on the `tkn_amt` directly after calling `userAmount(...)` (i.e., line 195 in this case). In short, Im suggesting we retain the original `validate_input(..)` which includes `tkn_amt`, but simply handle this after calling `userAmount(...)`.